src/core/server/saved_objects/service/lib/update_objects_spaces.test.ts

```diff
@@ -382,6 +389,146 @@ describe('#updateObjectsSpaces', () => {
+        expect(client.bulk).toHaveBeenCalledTimes(1);
+        expectBulkArgs({ action: 'update', object: { ...obj2, namespaces: [space3, space1] } });
+        expect(mockDeleteLegacyUrlAliases).not.toHaveBeenCalled();
+        expect(params.logger.error).not.toHaveBeenCalled();
+      });
+
+      it('does not delete aliases for objects that were removed from spaces but were also added to All Spaces (*)', async () => {
+        const space2 = 'space-to-remove';
+        const obj1 = { type: SHAREABLE_OBJ_TYPE, id: 'id-1', spaces: [space2] };
+
+        const objects = [obj1];
+        const spacesToAdd = [ALL_NAMESPACES_STRING];
+        const spacesToRemove = [space2];
+        const params = setup({ objects, spacesToAdd, spacesToRemove });
+        // this test case does not call mget
+        mockBulkResults({ error: false }); // result for obj1
+
+        await updateObjectsSpaces(params);
+        expect(client.bulk).toHaveBeenCalledTimes(1);
+        expectBulkArgs({
+          action: 'update',
+          object: { ...obj1, namespaces: [ALL_NAMESPACES_STRING] },
+        });
+        expect(mockDeleteLegacyUrlAliases).not.toHaveBeenCalled();
+        expect(params.logger.error).not.toHaveBeenCalled();
+      });
+
+      it('deletes aliases for objects that were removed from specific spaces using "deleteBehavior: exclusive"', async () => {
+        const space1 = 'space-to-remove';
+        const space2 = 'other-space';
+        const obj1 = { type: SHAREABLE_OBJ_TYPE, id: 'id-1', spaces: [space2] }; // will not be changed
+        const obj2 = { type: SHAREABLE_OBJ_TYPE, id: 'id-1', spaces: [space1, space2] }; // will be updated
+        const obj3 = { type: SHAREABLE_OBJ_TYPE, id: 'id-1', spaces: [space1] }; // will be deleted
+
+        const objects = [obj1, obj2, obj3];
+        const spacesToRemove = [space1];
+        const params = setup({ objects, spacesToRemove });
+        // this test case does not call mget
+        mockBulkResults({ error: false }, { error: false }); // result2 for obj2 and obj3
+
+        await updateObjectsSpaces(params);
+        expect(client.bulk).toHaveBeenCalledTimes(1);
+        expectBulkArgs(
+          { action: 'update', object: { ...obj2, namespaces: [space2] } },
+          { action: 'delete', object: obj3 }
+        );
+        expect(mockDeleteLegacyUrlAliases).toHaveBeenCalledTimes(2);
+        expect(mockDeleteLegacyUrlAliases).toHaveBeenNthCalledWith(
+          1, // the first call resulted in an error which generated a log message (see assertion below)
+          expect.objectContaining({
+            type: obj2.type,
+            id: obj2.id,
+            namespaces: [space1],
+            deleteBehavior: 'inclusive',
+          })
+        );
+        expect(mockDeleteLegacyUrlAliases).toHaveBeenNthCalledWith(
+          2,
+          expect.objectContaining({
+            type: obj3.type,
+            id: obj3.id,
+            namespaces: [space1],
```

Comment:
Could you add a test where the `namespaces` argument for each object is different?

e.g
```
obj1 : ns-1, ns-2, ns-3
obj2: ns-1, ns-2

spacesToRemove: ns-1, ns-3
```